docs/credential_plugins.md

```diff
@@ -0,0 +1,329 @@
+Credential Plugins
+==================
+
+By default, sensitive credential values (such as SSH passwords, SSH private
+keys, API tokens for cloud services) in AWX are stored in the AWX database
+after being encrypted with a symmetric encryption cipher utilizing AES-256 in
+CBC mode alongside a SHA-256 HMAC.
+
+Alternatively, AWX supports retrieving secret values from third-party secret
+management systems, such as HashiCorp Vault and Microsoft Azure Key Vault.
+These external secret values will be fetched on demand every time they are
+needed (generally speaking, immediately before running a playbook that needs
+them).
+
+Configuring Secret Lookups
+--------------------------
+
+When configuring AWX to pull a secret from a third party system, there are
+generally three steps.
+
+Here is an example of creating an (1) AWX Machine Credential with
+a static username, `example-user` and (2) an externally sourced secret from
+HashiCorp Vault Key/Value system which will populate the (3) password field on
+the Machine Credential.
+
+1.  Create the Machine Credential with a static username, `example-user`.
+
```

Comment:
Could you clarify what to expect in the case that a field is required?

In the case of this example, say that instead of the "Machine" type, I'm using "Amazon Web Services". I want to use hashicorp vault to store the `password` field.

For AWS credentials, `password` is a required field. So in Step (1) here, I cannot create the credential without providing a value for the password. Do I have to put in `"password": "foo"`, so that the API will let me save it so that I can continue through the next steps to link to the external secrets?

Reply:
Bleh, this is a good question.  We may need to update some of these to remove the `required` requirement (since you can't really apply the input source relationship until the object itself has been saved).

Reply:
Okay, addressed in https://github.com/ansible/awx/pull/3098/commits/7d4c32f38c0f1cfa155c80f18d284818ee68e797